src/tarski/syntax/terms.py

```diff
@@ -187,6 +198,167 @@ def is_syntactically_equal(self, other):
         # Else we just need to recursively check if all subterms are syntactically equal
         return all(x.is_syntactically_equal(y) for x, y in zip(self.subterms, other.subterms))
 
+
+class AggregateCompoundTerm(Term):
+    """
+        Combinations of a functional symbol, a set of bound variables and a terms.
+        Argument symbol needs to be an instance of Function of have
+        Function-like interface.
+    """
+
+    def __init__(self, operator, bound_vars, subterm: Term):
+        self.symbol = operator
+        self.bound_vars = bound_vars
+        self.subterm = subterm  # TODO: type checking?
+
+    @property
+    def language(self):
+        return self.subterm.language
+
+    @property
+    def sort(self):
+        return self.subterm.sort
+
+    def __str__(self):
+        return '{}_{{{}}}({})'.format(self.symbol, ','.join([str(x) for x in self.bound_vars]), str(self.subterm))
+
+    __repr__ = __str__
+
+    def __hash__(self):
+        return hash((self.symbol.symbol, tuple(x for x in self.subterms)))
```

Comment:
This should simply be `self.subterm`, I guess?
```suggestion
        return hash((self.symbol.symbol, self.subterm))
```

Reply:
Wow, I can suggest the code change in the comment :-)

Reply:
These new features on git are pretty amazing.

Reply:
Resolved with commit 222ac807c2b0b9801889118500cf4abfbce093ac